tests/__snapshots__/generator.spec.ts.snap

```diff
@@ -1,125 +1,215 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
-exports[`complex interpolation case snapshot 1`] = `
+exports[`generator complex interpolation case snapshot 1`] = `
 "/* eslint-disable */
 /* tslint:disable */
-export function LocaleKeys<R extends string>(t: (...args: unknown[]) => R) {
-  return {
+export type ILocaleKeys = {
     order: {
       shippingLabel: {
         customerDetailsCard: {
-          address: (data: Record<'firstName' | 'lastName' | 'addressLine1' | 'addressLine2' | 'city' | 'state' | 'zipCode' | 'country', unknown>) => t('order.shippingLabel.customerDetailsCard.address', data), /* {{firstName}} {{lastName}}, {{addressLine1, suffix ', '}}{{addressLine2, suffix ', '}}{{city, concat ', '}}{{state, suffix ' '}}{{zipCode, suffix ', '}}{{country}} */
+          address: (data: Record<'firstName' | 'lastName' | 'addressLine1' | 'addressLine2' | 'city' | 'state' | 'zipCode' | 'country', unknown>) => string, /* {{firstName}} {{lastName}}, {{addressLine1, suffix ', '}}{{addressLine2, suffix ', '}}{{city, concat ', '}}{{state, suffix ' '}}{{zipCode, suffix ', '}}{{country}} */
```

Comment:
I have one concern! 
and maybe it's only me doing it this way...
many times I search the content in localeKeys file and easily copy the their path from the actual function which makes it really easy while developing:
`t('order.shippingLabel.customerDetailsCard.address', data)`
now with the new approach this won't be possible.
option 1: It's not really important
option 2: we can add it as a comment before the value

It is just something i thought about, wanted to hear from you if you can relate, or we can just live without it

Reply:
> option 2: we can add it as a comment before the value

sounds good to me
will do